.github/workflows/warning_test.yml

```diff
@@ -0,0 +1,26 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+# More GitHub Actions for Azure: https://github.com/Azure/actions
+
+name: On pull request, build and show errors for warnings in project.
+
+on:
+  pull_request:
+    branches:
+      [ main ]
+
+jobs:
+  build:
+    name: Build with Warnings as Errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
+          include-prerelease: true
+
+      - name: Build with dotnet
+        run: dotnet build -warnaserror
```

Comment:
I think we can add this flag to the existing workflow (`test_on_push_bricks-and-hearts.yml`) as opposed to making a new one 🙂